### PR TITLE
docs: Add Paymaster section to the Transaction Component docs

### PR DESCRIPTION
**Pull request by cpcramer**
**What changed? Why?**

Add a section to the Transaction Component docs about getting the paymaster to work. 
- It's currently unclear how a developer set up their Paymaster & Bundler endpoint from the documentation
- https://linear.app/coinbase/issue/BOE-104/paymaster-set-up-clarification

Screenshot 2024-07-29 at 2 37 19 PM

**Notes to reviewers**

**How has it been tested?**

### CODE REVIEW

File: site/docs/pages/transaction/transaction.mdx [hunk 1]
```diff
@@ -85,19 +85,19 @@ export default function TransactionComponents() {
   );
 }
 ```
-
 <App>
   <TransactionWrapper>
     {({ address, contracts, onError }) => {
+      const capabilities = {
+        paymasterService: {
+          url: 'https://api.developer.coinbase.com/rpc/v1/base-sepolia/KkgONfLFgUZKI8nnrttLbI78U2vyhtXg',
```
Comment (Zizzamia) [inline site/docs/pages/transaction/transaction.mdx:93]: @cpcramer is this a real Key? like what happen when somoene will copy this?

File: site/docs/pages/transaction/transaction.mdx [hunk 2]
```diff
@@ -123,6 +123,63 @@ export default function TransactionComponents() {
   </TransactionWrapper>
 </App>
 
+## `Transaction` with Paymaster Gas Sponsorship Capabilities
+##### Obtain a Paymaster and Bundler endpoint from the [Coinbase Developer Platform Paymaster](https://portal.cdp.coinbase.com/products/bundler-and-paymaster).
+
+<img
+  alt="OnchainKit Paymaster and Bundler endpoint"
+  title="OnchainKit Paymaster and Bundler endpoint"
+  src="/assets/onchainkit-components-paymaster-endpoint.png"
```
Comment (Zizzamia) [inline site/docs/pages/transaction/transaction.mdx:132]: In the image we completelly cover the URL. I have a feeling people might not know that is an URL, maybe we should only cover the last part where the Key is exposed.

File: site/docs/pages/transaction/transaction.mdx [hunk 3]
```diff
@@ -123,6 +123,63 @@ export default function TransactionComponents() {
...
+  width="auto"
+/>
+```tsx
+// ommited for brevity
+<Transaction
+  address={address}
+  contracts={contracts}
+  capabilities={{ // [!code focus]
+    paymasterService: { // [!code focus]
+      url: process.env.PAYMASTER_AND_BUNDLER_ENDPOINT, // [!code focus]
```
Comment (Zizzamia) [inline site/docs/pages/transaction/transaction.mdx:142]:
```
process.env.PAYMASTER_AND_BUNDLER_ENDPOINT
```
is this simple enough to understand? I believe we approach this in a different way in the Getting Started, or no?

File: site/docs/pages/transaction/transaction.mdx [hunk 4]
```diff
@@ -123,6 +123,63 @@ export default function TransactionComponents() {
...
+    {({ address, contracts, onError }) => {
+      if (address) {
+        return (
+          <Transaction 
+            address={address} 
+            contracts={contracts} 
+            onError={onError} 
+            capabilities={{
+              paymasterService: {
+                url: 'https://api.developer.coinbase.com/rpc/v1/base/KkgONfLFgUZKI8nnrttLbI78U2vyhtXg',
```
Comment (Zizzamia) [inline site/docs/pages/transaction/transaction.mdx:161]: Whos key are we using here? Should this be store in Vercel?

File: site/docs/pages/transaction/transaction.mdx (hunk 3 above)
Comment (cpcramer) [reply site/docs/pages/transaction/transaction.mdx:142]: In the getting started guide we had an additional step that showed how to add the environment variable to a .env file

File: site/docs/pages/transaction/transaction.mdx (hunk 4 above)
Comment (cpcramer) [reply site/docs/pages/transaction/transaction.mdx:161]: This is my CDP key - Adding to Vercel

**Comment by Zizzamia** [comment]
@cpcramer make sure to rebase

File: site/docs/pages/transaction/transaction.mdx (hunk 2 above)
Comment (cpcramer) [reply site/docs/pages/transaction/transaction.mdx:134]: updated

**Review by Zizzamia — APPROVED** [review]

**Pull request merged**